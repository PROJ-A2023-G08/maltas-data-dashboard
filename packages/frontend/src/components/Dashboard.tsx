import { useTranslation } from "next-i18next";

const Dashboard = () => {
  const { t } = useTranslation("common");

  return <h1>{t("sidebar.Dashboard")}</h1>;
};

export default Dashboard;
