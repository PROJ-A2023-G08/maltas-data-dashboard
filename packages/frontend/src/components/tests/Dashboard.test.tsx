import React from "react";
import renderer from "react-test-renderer";
import Dashboard from "../Dashboard";

// Mock the useTranslation hook from i18next to return a translation function
jest.mock('next-i18next', () => ({
    useTranslation: jest.fn().mockReturnValue({
        // @ts-ignore
        t: str => str,
    }),
}));

describe("Dashboard", () => {
    it("renders correctly", () => {
        const tree = renderer.create(<Dashboard  />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});