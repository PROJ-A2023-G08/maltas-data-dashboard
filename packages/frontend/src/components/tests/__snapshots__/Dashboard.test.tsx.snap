// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders correctly 1`] = `
<h1>
  sidebar.Dashboard
</h1>
`;
